proxy/src/telemetry/metrics/mod.rs

```diff
@@ -159,26 +159,26 @@ impl Metrics {
     }
 
     fn request_total(&mut self,
-                     labels: &Arc<RequestLabels>)
+                     labels: RequestLabels)
```

Comment:
We could avoid cloning in the majority of cases if we didn't use the `entry` api. All lookups for existing keys can be done without cloning.

Reply:
That's true; I  think we can get away with at least only using the metrics API when a request/response is _opened_, and use `.get_mut(...).expect(...)` on close events, since we should already have a key in the map. I'll make that change in a separate PR.